src/prayer/AnswerPrayerForm/AnswerPrayerFormConnected.js

```diff
@@ -19,33 +38,71 @@ class AnswerPrayerFormConnected extends React.Component {
             currentUser: { profile: { photo = { uri: '' } } = {} } = {},
           } = {},
         }) => (
-          <Mutation mutation={ANSWER_PRAYER}>
-            {(answerPrayer) => (
-              <AnswerPrayerForm
-                loading={profileLoading}
-                onSubmit={(values) => {
-                  answerPrayer({
-                    variables: {
-                      id: values.id,
-                      answer: values.answer,
-                    },
-                    refetchQueries: () => [
-                      { query: GET_PRAYER_FEED, variables: { type: 'USER' } },
-                    ],
-                  });
-                  this.props.navigation.pop();
-                }}
-                avatarSource={photo}
-                {...this.props}
-                onClose={() => this.props.navigation.pop()}
-                title={"Celebrate God's faithfulness"}
-                prayerId={this.props.navigation.getParam('prayerId', '')}
-                prayerText={this.props.navigation.getParam('prayerText', '')}
-                prayerAnswer={this.props.navigation.getParam(
-                  'prayerAnswer',
-                  ''
+          <Mutation mutation={INTERACT_WITH_PRAYER}>
+            {(interactWithPrayer) => (
+              <Mutation mutation={ANSWER_PRAYER}>
+                {(answerPrayer) => (
+                  <AnswerPrayerForm
+                    loading={profileLoading}
+                    onSubmit={(values) => {
+                      answerPrayer({
+                        variables: {
+                          id: values.id,
+                          answer: values.answer,
+                        },
+                        refetchQueries: () => [
+                          {
+                            query: GET_PRAYER_FEED,
+                            variables: { type: 'USER' },
+                          },
+                        ],
+                      });
+                      this.props.navigation.pop();
+                    }}
+                    avatarSource={photo}
+                    {...this.props}
+                    onClose={() => this.closePrayer()}
+                    title={"Celebrate God's faithfulness"}
+                    prayerId={this.prayerId}
+                    prayerText={this.props.navigation.getParam(
+                      'prayerText',
+                      ''
+                    )}
+                    prayerAnswer={this.prayerAnswer}
```

Comment:
is there a reason you're defining some of these as separate functions and variables?

Reply:
So I did that because I use `prayerId` and `prayerAnswer` a little further down. So instead of calling the `getParam` function again I just made one function for each of them that would set it.